src/spatialdata/models/models.py

```diff
@@ -182,7 +182,7 @@ def parse(
                 ) from e
 
         # finally convert to spatial image
-        data = to_spatial_image(array_like=data, dims=cls.dims.dims, **kwargs)
+        data = DataArray(to_spatial_image(array_like=data, dims=cls.dims.dims, **kwargs))
```

Comment:
to_spatial_image returns a `DataArray` and does some internal validation so this should be reverted.
```suggestion
        data = to_spatial_image(array_like=data, dims=cls.dims.dims, **kwargs)
```